substrate/frame/support/src/traits/tokens/fungibles/imbalance.rs

```diff
@@ -158,6 +157,58 @@ impl<
 	}
 }
 
+impl<
+		A: AssetId,
+		B: Balance,
+		OnDrop: HandleImbalanceDrop<A, B>,
+		OppositeOnDrop: HandleImbalanceDrop<A, B>,
+	> PartialEq for Imbalance<A, B, OnDrop, OppositeOnDrop>
+{
+	fn eq(&self, other: &Self) -> bool {
+		self.amount.eq(&other.amount) && self.asset.eq(&other.asset)
+	}
+}
+
+impl<
+		A: AssetId,
+		B: Balance,
+		OnDrop: HandleImbalanceDrop<A, B>,
+		OppositeOnDrop: HandleImbalanceDrop<A, B>,
+	> Eq for Imbalance<A, B, OnDrop, OppositeOnDrop>
+{
+}
```

Comment:
You should be able to just use `EqNoBound` and `PartialEqNoBound` for this?

Reply:
Nice, I was not aware of them.